app/src/main/java/com/android/voyageur/ui/overview/Overview.kt

```diff
@@ -171,57 +129,190 @@ fun OverviewScreen(
+ * state of the app.
+ *
+ * @param isLoading Boolean indicating whether data is currently being loaded
+ * @param trips List of all available trips
+ * @param searchQuery Current search query for filtering trips
+ * @param padding PaddingValues to apply to the content
+ * @param tripsViewModel ViewModel containing trips-related state and logic
+ * @param navigationActions Actions to handle navigation between screens
+ * @param userViewModel ViewModel containing user-related state and logic
+ */
+@Composable
+private fun OverviewContent(
+    isLoading: Boolean,
+    trips: List<Trip>,
+    searchQuery: String,
+    padding: PaddingValues,
+    tripsViewModel: TripsViewModel,
+    navigationActions: NavigationActions,
+    userViewModel: UserViewModel
+) {
+  if (isLoading) {
+    CircularProgressIndicator(modifier = Modifier.testTag("loadingIndicator"))
+    return
+  }
+
+  Column(
+      modifier = Modifier.padding(padding).testTag("overviewColumn"),
+  ) {
+    when {
+      trips.isEmpty() -> EmptyTripsMessage()
+      else ->
+          TripsList(
+              trips = trips,
+              searchQuery = searchQuery,
+              tripsViewModel = tripsViewModel,
+              navigationActions = navigationActions,
+              userViewModel = userViewModel)
+    }
+  }
+}
+
+/**
+ * Composable that displays a message when no trips are available.
+ *
+ * Shows a centered message prompting the user to create their first trip.
+ */
+@Composable
+private fun EmptyTripsMessage() {
+  Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+    Text(
+        modifier = Modifier.testTag("emptyTripPrompt"),
+        text = stringResource(R.string.empty_trip_prompt),
+    )
+  }
+}
+
+/**
+ * Composable that renders the list of trips.
+ *
+ * Displays a scrollable list of trip items, filtered according to the search query. Shows a "no
+ * results" message when the search yields no matches.
+ *
+ * @param trips List of all available trips
+ * @param searchQuery Current search query for filtering trips
+ * @param tripsViewModel ViewModel containing trips-related state and logic
+ * @param navigationActions Actions to handle navigation between screens
+ * @param userViewModel ViewModel containing user-related state and logic
+ */
+@Composable
+private fun TripsList(
+    trips: List<Trip>,
+    searchQuery: String,
+    tripsViewModel: TripsViewModel,
+    navigationActions: NavigationActions,
+    userViewModel: UserViewModel
+) {
+  val filteredTrips = filterTrips(trips, searchQuery)
+
+  if (searchQuery.isNotEmpty() && filteredTrips.isEmpty()) {
+    NoResultsFound(modifier = Modifier.testTag("noSearchResults"))
+    return
+  }
+
+  LazyColumn(
+      verticalArrangement = Arrangement.spacedBy(10.dp, Alignment.Top),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      modifier = Modifier.fillMaxSize().testTag("lazyColumn")) {
+        filteredTrips.forEach { trip ->
+          item {
```

Comment:
Here I think you can use the `items` function as it looks a bit "cleaner", but it's not a big deal.

Reply:
Thank you for the suggestion! I agree with you that it would make the code cleaner :)